app/src/androidTest/java/com/epfl/drawyourpath/mainpage/fragments/SettingsFragmentTest.kt

```diff
@@ -0,0 +1,98 @@
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.authentication.MOCK_FORCE_SIGNED
+import com.epfl.drawyourpath.authentication.USE_MOCK_AUTH
+import com.epfl.drawyourpath.preferences.ModifyPasswordFragment
+import org.hamcrest.TypeSafeMatcher
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+class ToastMatcher : TypeSafeMatcher<Root>() {
+
+    override fun describeTo(description: org.hamcrest.Description?) {
+        description?.appendText("is toast");
+    }
+
+    override fun matchesSafely(root: Root): Boolean {
+        val type = root.windowLayoutParams.get().type;
+        if ((type == WindowManager.LayoutParams.TYPE_TOAST)) {
+            val windowToken = root .decorView.windowToken;
+            val appToken = root .decorView.applicationWindowToken;
+            if (windowToken == appToken) {
+                return true;
+            }
+        }
+        return false;
+    }
+}
+
+
+@RunWith(AndroidJUnit4::class)
+class SettingsFragmentTest {
+
+    private fun toastHasText(text: String) {
+        onView(withText("Not signed in."))
+    }
+
+    @Test
+    fun userCantUpdatePasswordWhenNotSigned() {
+        val scenario = launchFragmentInContainer<ModifyPasswordFragment>(bundleOf(
+            Pair(USE_MOCK_AUTH, true)
+        ))
+
+        onView(withId(R.id.BT_Apply)).perform(click())
+        toastHasText("Not signed in.")
+    }
+
+    @Test
+    fun emptyPasswordFailsToUpdate() {
+        val scenario = launchFragmentInContainer<ModifyPasswordFragment>(bundleOf(
+            Pair(USE_MOCK_AUTH, true),
+            Pair(MOCK_FORCE_SIGNED, true)
+        ))
+
+        onView(withId(R.id.BT_Apply)).perform(click())
+        toastHasText("Password is empty")
+    }
+
+    @Test
+    fun mismatchingPasswordFailsToUpdate() {
+        val scenario = launchFragmentInContainer<ModifyPasswordFragment>(bundleOf(
+            Pair(USE_MOCK_AUTH, true),
+            Pair(MOCK_FORCE_SIGNED, true)
+        ))
+
+        onView(withId(R.id.ET_Password)).perform(clearText(),
+            typeText("NewP4ss\\/\\/0rd"))
+        onView(withId(R.id.ET_PasswordRepeat)).perform(clearText(),
+            typeText("NewP3ss\\/\\/0rdfoobar"))
+        toastHasText("Passwords don't match")
+    }
+
+    @Test
+    fun validPasswordPassesUpdate() {
+        val scenario = launchFragmentInContainer<ModifyPasswordFragment>(bundleOf(
+            Pair(USE_MOCK_AUTH, true),
+            Pair(MOCK_FORCE_SIGNED, true)
+        ))
+
+        onView(withId(R.id.ET_Password)).perform(clearText(),
+            typeText("NewP4ss\\/\\/0rd"))
+        onView(withId(R.id.ET_PasswordRepeat)).perform(clearText(),
+            typeText("NewP4ss\\/\\/0rd"))
+        toastHasText("Passwords updated")
+    }
```

Comment:
Suggestion: Change the test names in order to better illustrate the expected behavior. For ex: userCantUpdatePasswordWhenNotSigned can be changed to shouldShowErrorWhenUserNotSignedIn.

Reply:
I've tried to come up with better names for the test. Let me know if it still need some work.